src/views/Submit.vue

```diff
@@ -266,11 +289,31 @@ export default {
 			this.$refs.submitButton.click()
 		},
 
+		/*
+		 * Methods for catching unwanted unload events
+		 */
+		beforeWindowUnload(e) {
+			if (this.confirmLeaveForm()) {
+				// Cancel the window unload event
+				e.preventDefault()
+				e.returnValue = ''
+			}
+		},
+
+		confirmLeaveForm() {
+			return (
+				!this.submitForm
+				&& Object.keys(this.answers).length > 0
+				&& confirm(t('forms', 'You have unsaved changes! Do you still want to leave?'))
+			)
```

Comment:
See above.
It sounds like leaving is confirmed, so we should return whenever the navigation should continue and false if we want to cancel the navigation.

```suggestion
		/**
		Check if the form contains unsaved changes, returns true if the the form can be leaved safely, false if the navigation should be canceled.
		 */
		confirmLeaveForm() {
			return (
				this.submitForm
				|| Object.keys(this.answers).length === 0
				|| confirm(t('forms', 'You have unsaved changes! Do you still want to leave?'))
			)
```